core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -279,7 +280,8 @@ public NativeImageBuildItem build(NativeConfig nativeConfig, NativeImageSourceJa
         }
     }
 
-    private static NativeImageBuildRunner getNativeImageBuildRunner(NativeConfig nativeConfig, Path outputDir) {
+    private static NativeImageBuildRunner getNativeImageBuildRunner(NativeConfig nativeConfig, Path outputDir,
+            String nativeImageName) {
         boolean isContainerBuild = nativeConfig.containerRuntime.isPresent() || nativeConfig.containerBuild;
```

Comment:
I think `nativeConfig.remoteContainerBuild` should be added as well.

Reply:
Good point! My intention was that the `nativeConfig.remoteContainerBuild` is actually only respected when `nativeConfig.remoteContainer` is `true` and it would simply be ignored when `nativeConfig.remoteContainer` is `false`. However, thinking about it, this might be confusing behavior. I added it to the condition above as you suggest, such that it is enough to set `nativeConfig.remoteContainerBuild` to `true` to trigger a remote container build.

The question then is, what runner should be chosen when both `nativeConfig.remoteContainer` and `nativeConfig.remoteContainerBuild` are set to `true`? The current code would choose the remote container build in that case, which I think is fine.

Reply:
> The current code would choose the remote container build in that case, which I think is fine.

Yes, that's what I'd expect as a user.